prinz_pmml/src/main/scala/pl/touk/nussknacker/prinz/pmml/repository/PMMLModelRepository.scala

```diff
@@ -1,39 +1,32 @@
 package pl.touk.nussknacker.prinz.pmml.repository
 
 import com.typesafe.scalalogging.LazyLogging
-import pl.touk.nussknacker.prinz.model.repository.{ModelRepository, ModelRepositoryException}
+import pl.touk.nussknacker.prinz.model.repository.{ModelRepository}
 import pl.touk.nussknacker.prinz.model.ModelName
 import pl.touk.nussknacker.prinz.pmml.PMMLConfig
 import pl.touk.nussknacker.prinz.pmml.model.PMMLModel
-import pl.touk.nussknacker.prinz.pmml.repository.PMMLModelRepository.PMML_FILE_EXTENSION
+import pl.touk.nussknacker.prinz.util.repository.{ModelPayload, RepositoryClient}
 
-import java.net.URL
-
-
-abstract class PMMLModelRepository(implicit protected val config: PMMLConfig)
+class PMMLModelRepository(implicit config: PMMLConfig)
   extends ModelRepository with LazyLogging {
 
+  private val client = new RepositoryClient
+  private val uri = config.modelsDirectory
+
   override def listModels: RepositoryResponse[List[PMMLModel]] =
-    Right[ModelRepositoryException, List[PMMLModel]](readURL.map(PMMLModel(_)).toList)
+    client.listModelFiles(uri).right.map(it => it.map(p => PMMLModel(mapPayload(p))).toList)
 
   override def getModel(name: ModelName): RepositoryResponse[PMMLModel] =
-    Right[ModelRepositoryException, PMMLModel](readURL.filter(p => p.name == name.toString)
-      .map(PMMLModel(_)).head)
+    client.listModelFiles(uri)
+      .right.map(it => it.map(mapPayload).filter(p => p.name == name.toString))
+      .map(it => PMMLModel(it.head))
 
-  private def readURL: Iterable[PMMLModelPayload] = {
-    val url = config.modelsDirectory
-    logger.info(s"Protocol used in PMMLModelRepository is ${url.getProtocol}")
-    readPMMFilesData(url, config)
-  }
-
-  protected def readPMMFilesData(url: URL, config: PMMLConfig): Iterable[PMMLModelPayload]
-
-  protected def isPMMLFile(fileName: String): Boolean = fileName.endsWith(PMML_FILE_EXTENSION)
+  private def mapPayload(payload: ModelPayload): PMMLModelPayload =
+    PMMLModelPayload(payload, client.openModelFile(payload.path), config.fileExtension)
 }
 
 object PMMLModelRepository {
 
+  //TODO: I think this also should be in config
```

Comment:
do this also but imo don't place it to common config trait (it will be somehow duplicated code but he semantics of this trait RepositoryClientConfig has nothing common with the version encoded in filename so ti should be separated)